src/ctl.c

```diff
@@ -2289,6 +2291,40 @@ thread_tcache_enabled_ctl(tsd_t *tsd, const size_t *mib,
 	return ret;
 }
 
+static int
+thread_tcache_max_ctl(tsd_t *tsd, const size_t *mib,
+    size_t miblen, void *oldp, size_t *oldlenp, void *newp,
+    size_t newlen) {
+	int ret;
+	size_t oldval;
+
+	/* pointer to tcache_t always exists even with tcache disabled. */
+	tcache_t *tcache = tsd_tcachep_get(tsd);
+	assert(tcache != NULL);
+	oldval = tcache_max_get(tcache);
+	READ(oldval, size_t);
+
+	if (newp != NULL) {
+		if (newlen != sizeof(size_t)) {
+			ret = EINVAL;
+			goto label_return;
+		}
+		size_t new_tcache_max = oldval;
+		WRITE(new_tcache_max, size_t);
+		if (new_tcache_max > TCACHE_MAXCLASS_LIMIT) {
+			new_tcache_max = TCACHE_MAXCLASS_LIMIT;
+		}
+		new_tcache_max = sz_s2u(new_tcache_max);
```

Comment:
nit: let's switch this with the if branch above, to make sure the cap after all potential increases.

Reply:
This cannot be switched, because `sz_s2u` causes overflow in 32-bit systems. The order won't affect the outcome as well because the `if` can only reduce the value of `new_tcache_max` and `sz_s2u` won't make the value increase.